Simplified/NYPLMigrationManager.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+/**
+Manages data migrations as they are needed throughout the app's life
+
+App version is cached in UserDefaults and last cached value is checked against current build version
+and updates are applied as required
+*/ 
+class MigrationManager: NSObject {
+  @objc static func migrate() {
+    // Fetch target version
+    let targetVersion = Bundle.main.infoDictionary!["CFBundleShortVersionString"] as! String
+
+    // Fetch and parse app version
+    let appVersion = NYPLSettings.shared().appVersion ?? ""
+    let appVersionTokens = appVersion.filter({ $0.isNumber || $0 == "." }).split(separator: ".").map({ Int($0)! })
+    
+    // Run through migration stages
+    if versionComparator(appVersionTokens, [3, 2, 0]) { // v3.2.0
+      migrate1();
+    }
+
+    // Migrate Network Queue DB
+    NetworkQueue.sharedInstance.migrate()
+
+    // Update app version
+    NYPLSettings.shared().appVersion = targetVersion
+  }
+
+  // Less-than comparator operation
+  private static func versionComparator(_ a: [Int], _ b:[Int]) -> Bool {
+    var i = 0
+    while i < a.count && i < b.count {
+      if (a[i] < b[i]) {
+        return true
+      }
+      i += 1
+    }
+    return a.count < b.count && b[i] > 0
+  }
+
+  // v3.2.0
+  private static func migrate1() -> Void {
+    // Build account map
+    var accountMap = [Int: String]()
+    if let accountsUrl = Bundle.main.url(forResource: "Accounts", withExtension: "json") {
+      do {
+        let accountsData = try Data(contentsOf: accountsUrl)
+        let accountsDataObj = try JSONSerialization.jsonObject(with: accountsData, options: .allowFragments)
+        if let accountsDataArray = accountsDataObj as? [[String: AnyObject]] {
+          for jsonDict in accountsDataArray {
+            if let numericIdNullable = jsonDict["id_numeric"] {
```

Comment:
You could do this line and the line after in one:
`if let numericIdNullable = jsonDict["id_numeric"] as? Int {`
Or if you want it to explode if it's not an `Int`, then just:
`let numericId = jsonDict["id_numeric"] as! Int`